subsys/bluetooth/host/hci_core.c

```diff
@@ -137,9 +137,13 @@ static struct cmd_data cmd_data[CONFIG_BT_HCI_CMD_COUNT];
 #define cmd(buf) (&cmd_data[net_buf_id(buf)])
 #define acl(buf) ((struct acl_data *)net_buf_user_data(buf))
 
-void bt_hci_cmd_data_state_set(struct net_buf *buf,
-			       struct bt_hci_cmd_state_set *state)
+void bt_hci_cmd_state_set_init(struct net_buf *buf,
```

Comment:
I thought keeping the _init variant made most sense, since it is all just initializing the state change when the command completes, now it just includes assigning the pointer from the buf to the state change.

Also the _data_ part in `bt_hci_cmd_data_state_set` was there just to avoid a name conflict with the struct itself, which is another reason why I preferred the _init name.